### PR TITLE
Dashboard gets data via ES.

### PR DESCRIPTION
**Pull request by elcct**
As in the title.

On my laptop modified dashboard endpoint returns data @ ~65ms vs. ~2s when using the Database.

I checked it with latest Frontend and it appears to be working.

### CODE REVIEW

File: datahub/dashboard/serializers.py [hunk 1]
```diff
@@ -1,31 +0,0 @@
...
-        model = Interaction
-        depth = 2
-        fields = '__all__'
-
-
-class IntelligentHomepageSerializer(serializers.Serializer):
-    """Intelligent homepage serializer."""
-
-    interactions = InteractionSerializer(many=True)
-    contacts = NestedContactSerializer(many=True)
```
Comment (reupen) [inline datahub/dashboard/serializers.py]: Was the only usage of NestedContactSerializer, so can remove it now from datahub/company/serializers.py.

Comment (elcct) [reply datahub/dashboard/serializers.py]: Done, thanks!

File: datahub/search/dashboard/views.py [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+        serializer.is_valid(raise_exception=True)
+        limit = serializer.validated_data['limit']
+
+        interactions_query = get_search_by_entity_query(
+            term='',
+            entity=Interaction,
+            filters={
+                'dit_adviser.id': user.id,
+            },
+            field_order='created_on:desc',
```
Comment (reupen) [inline datahub/search/dashboard/views.py]: Before there was logic to exclude records with a null `created_on` date. I can see this puts nulls last, but would suggest keeping the logic to exclude them (unless there's a reason for keeping them).

File: datahub/search/dashboard/views.py [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+        )
+        contacts = limit_search_query(
+            contacts_query,
+            offset=0,
+            limit=limit,
+        ).execute()
+
+        response = {
+            'interactions': [interaction.to_dict() for interaction in interactions.hits],
+            'contacts': [contact.to_dict() for contact in contacts],
```
Comment (reupen) [inline datahub/search/dashboard/views.py]: Any reason for using .hits for interactions but not contacts?

File: datahub/search/dashboard/serializers.py [hunk 4]
```diff
@@ -0,0 +1,7 @@
+from rest_framework import serializers
+
+
+class LimitParamSerializer(serializers.Serializer):
```
Comment (reupen) [inline datahub/search/dashboard/serializers.py]: Would suggest just calling this HomepageSerializer as it looks pretty specific to the home page (with default=5).

File: datahub/search/dashboard/views.py (hunk 2 above)
Comment (elcct) [reply datahub/search/dashboard/views.py]: Good catch :) Simply forgot about it. Fixed.

File: datahub/search/dashboard/views.py (hunk 3 above)
Comment (elcct) [reply datahub/search/dashboard/views.py]: No reason, somehow lost it. Fixed, thanks :)

File: datahub/search/dashboard/serializers.py (hunk 4 above)
Comment (elcct) [reply datahub/search/dashboard/serializers.py]: Good idea. Renamed.

**Pull request merged**